src/main/kotlin/com/patxi/poetimizely/generator/FeaturesGenerator.kt

```diff
@@ -0,0 +1,109 @@
+package com.patxi.poetimizely.generator
+
+import com.optimizely.ab.Optimizely
+import com.patxi.poetimizely.optimizely.Feature
+import com.squareup.kotlinpoet.*
+import java.io.StringWriter
+
+/**
+ * The functionality to parse features to type safe code
+ * is designed and documented in https://docs.google.com/document/d/1XJap6OKnzAM-C4GoL1Ubs2zKRzJSlIdzcvddOOE2Z7Q.
+ */
+class FeaturesGenerator(private val packageName: String = "") {
+    fun build(features: List<Feature>): String =
+        with(
+            FileSpec.builder(packageName, "Features")
+                .addType(featuresEnumTypeSpec(features))
+                .addType(featuresClientTypeSpec(packageName))
+                .build()
+        ) {
+            return StringWriter().also { appendable: Appendable ->
+                this.writeTo(appendable)
+            }.toString()
+        }
+}
+
+/**
+ * According to the Optimizely Docs (https://docs.developers.optimizely.com/full-stack/docs/create-feature-flags)
+ * a feature key has the following restrictions:
+ * - unique
+ * - alphanumeric characters
+ * - hyphens
+ * - underscores
+ * - max 64 chars
+ * - NO spaces
+ *
+ * The key will be used "to determine whether the feature is on or off in your code".
+ */
+private fun String.optimizelyFeatureKeyToEnumConstant() =
+    split("-", "_")
+        .joinToString("_") { it.trim().toUpperCase() }
```

Comment:
I think it is a good practice to type the return of functions that are not very trivial.
```suggestion
private fun String.optimizelyFeatureKeyToEnumConstant(): String =
    split("-", "_").joinToString("_") { it.trim().toUpperCase() }
```